### DISCUSSION TITLE
How to determine "write_slf_header" header variable "seek_head"

### DISCUSSION THREAD

**Discussion by KayShelton** [tpilz/telemac#2] [Q&A] [answered]
I am trying to write data to a .slf file using the "write_slf_variable" function. Before doing this, I need to create the file with "write_slf_header", providing the header information describing the contents of the file. One list value needed in the header is the "seek_head" effectively giving the length of the header. The documentation on this says this can be inferred from "read_slf_header" but as I am writing a new header with new information, I don't now how to specify the new seek_head. Please can you provide information on how I should determine the value for this?

**Comment by tpilz** [comment]
For writing you actually don't need the `seek_head` information. It should work just leaving it out. I admit that the function's doc is a bit unspecific here.

However, note that the `write_slf_*` functions are low-level functions. It is usually much more convenient to use the `geo()` or `write_results()` function which are basically wrappers around the `write_slf_*` functions.

**Comment by KayShelton** [comment]
Thanks @tpilz , good to know. I am trying to meteorological data (winds and pressure from ERA5) to the node points on a mesh for use by a colleague to run Telemac - I won't be running the model myself (and have no experience running it). They want all the met data in a single slf file as an input, so I figured this was the best way to go about it. Would you suggest otherwise?

**Comment by tpilz** [comment, marked as answer]
This depends on how these meteorological data have to be provided -- I have no experience here. When they have to be provided along with the mesh in the geometry file you can use function `cas()` as shown in [this vignette](https://cran.r-project.org/web/packages/telemac/vignettes/t2d_rainfall_runoff.html#TELEMAC-2D_initialisation) for CN and DEM elevation when using the rainfall-runoff module. If it works differently for the module you (or your colleague) are using you are maybe right to go with the `write_self_*` functions.

**Comment by KayShelton** [comment]
Much appreciated @tpilz. I managed to get it all sorted with the write_self_* functions, and they seem to read back in correctly with the read_slf_* functions. Thanks for your confirmation on the requirements.
